### DISCUSSION TITLE
How do I go and label the throat in an area?

### DISCUSSION THREAD

**Discussion by Xknightfyy** [PMEAL/OpenPNM#2769] [Q&A]
Dear Professor Jeff Gostick,

Sorry to bother you again, but I can't solve this problem, hope to get your help.I tried using pore.coords to define a throat in an area but I failed, can you tell me what I should do to label a throat in an area? Take the liberty to bother you, hope to get your help, thank you!

**Comment by jgostick** [comment]
I'm not sure what you mean.  I think you're referring to a prevous discussion thread, but can't remember which one.  

---
**Your question either contains too much or too little information**

Please re-ask your question by including a "minimal working example" so that someone (i.e., a dev or another user) can easily reproduce your problem or situation.  For instance, you might create a 4x4x1 cubic network, and use it to illustrate the problem your having. This way we can digest your conditions and reproduce your problem without needing access to your large images or data files. 

It's also important that any code snippet you upload contains only the steps needed to reproduce the problem.  It is time consuming and inefficient for someone else to dive into complex code before they can even begin to answer the question.  For instance if you are having problems with algorithm X, please don't include algorithms Y and Z in your script. Crafting a small working example that illustrates your problem may take you some time, but remember that wading through your large and messy example takes up our time. For example, posting an entire script into the comment box and asking someone why it doesn't work is not OK. 

On the other hand, it is also necessary to provide enough information that someone can appreciate what your trying to ask. In this regard, some code snippets are important, just remember to keep them short and to the point. [^1]

[^1]: This long response was 'auto-generated' using the [saved replies](https://docs.github.com/en/get-started/writing-on-github/working-with-saved-replies/using-saved-replies) feature.

**Comment by Xknightfyy** [comment]
Hello professor, the code is as follows, I can't define the throat in the lower half of the area, it can only show half of it, it can't show all the yelling channels in the lower half area, is this because of the coordinates of the center of mass of the throat ? So how should I define it?

import numpy as np
import openpnm as op
op.visualization.set_mpl_style()
np.random.seed(10)
np.set_printoptions(precision=5)

pn = op.network.Cubic(shape=[8, 8, 8], spacing=1e-6)
pn.add_model_collection(op.models.collections.geometry.spheres_and_cylinders)
pn.regenerate_models()

ax = op.visualization.plot_coordinates(pn)
ax = op.visualization.plot_connections(pn, ax=ax)
Ts =  pn['pore.coords'][:, 2] < 0.5 * pn['pore.coords'][:, 2].max()
pn.set_label(throats=Ts, label='Zone')

ax = op.visualization.plot_coordinates(network=pn, pores=pn.pores('bottom'), c='r', s=50)
ax = op.visualization.plot_coordinates(network=pn, pores=pn.pores('bottom'), c='grey', ax=ax)
op.visualization.plot_connections(network=pn, throats=pn.throats('Zone'), ax=ax);



<img width="961" alt="ee2834078bbe14ba6e495450b70c103" src="https://github.com/PMEAL/OpenPNM/assets/120148734/a30cbe02-0da4-4b27-8576-be53d4e322e3">

**Reply by jgostick** [reply]
Thank you for such as concise question...I was able to see the problem with just a few moments of looking.  your issue is this line `Ts = pn['pore.coords'][:, 2] < 0.5 * pn['pore.coords'][:, 2].max()`.  This is is not finding throats, it is finding pores whose coordinates match those conditions.  To find the throats connected to those pores you need to do:

```
Ps = pn['pore.coords'][:, 2] < 0.5 * pn['pore.coords'][:, 2].max()
Ts = pn.find_neighbor_throats(Ps)
```

**Reply by Xknightfyy** [reply]
Thanks for the guidance, it was very helpful to me!!!
